src/components/SelectCol.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react';
+
+class SelectCol extends React.Component {
+    constructor(props) {
+        super(props);
+        this.getCheckedValues = this.getCheckedValues.bind(this);
+        this.state = {
+            languages: [ "Arabic", "Serbian", "Croatian", "Russian", "German", "Hebrew", "French",
+                "Hungarian", "Slovak", "Spanish", "Portugues", "Turkce", "Greek", "Romanian"],
+            defaultLans: this.props.default
+        }
+    }
+
+    getCheckedValues(index){ //Get the selected Value and pass to other component
+        //toggle the default languages
+        if (index == null) {
+        }else {
+            const newLans = this.state.defaultLans.slice() //copy the array
+            newLans[index].value = !this.state.defaultLans[index].value;  //execute the manipulations
+            this.setState({ defaultLans: newLans }) //set the new state
+        }
+
+        const values = [];
+        const boxes = document.getElementsByClassName("custom-control-input");
+        for(let i=0; i< boxes.length; i++){
+            if(boxes[i].checked){
+               values.push(boxes[i].id);
+            }
+        }
+
+        //pass the values to the searchPage component
+        this.props.parentCallback(values);
+    }
+
+    //handle Event functions
+    openForm(){
+        const x = document.getElementById("language-options");
+        if (x.style.display === "none") {
+            x.style.display = "block";
+        } else {
+            x.style.display = "none";
+        }
+
+        document.addEventListener('click',function(e){
+            x.style.display = "none";
+        });
+    }
+
+    handleClick(e){ //stop propagation of the react
+        e.stopPropagation();
+        e.nativeEvent.stopImmediatePropagation();
+    }
+
+    render() {
+        console.log( "The Select Col render function gets called" );
+
+        //Style will move to css file later
+        const buttonStyle ={
+            borderRadius: "50%",
+            border:"none",
+            position:"fixed",
+            top: "8px",
+            right: "16px"
+        }
+
+        const cardStyle= {
+            width: "10rem",
+            display: "none",
+            paddingLeft: "10px",
+            paddingBottom:"10px",
+            position:"fixed",
+            top: "45px",
+            right: "16px",
+            //overflow:"auto"
+        }
+
+        const defaultLanguages = this.props.default.map(( language, index) =>
+            <div className="custom-control custom-checkbox">
+                <input type="checkbox" className="custom-control-input" id={ language.name } checked={ language.value }
```

Comment:
this checked thing actually causes a warning in the browser. I am thinking, should we use the onChange function instead of onClicked to get rid of the warning?

Reply:
I have no idea about this. And I don't think the warning is related to the onClick or onChange

Reply:
emmm, the warning says something about the onChange function, so maybe see if using onChange can solve it. There are also other warnings there, can you have a look at them?

Reply:
The warning for the button is solved. However, the first warning, the index can't be the same. It's from 0, 1, 2, 3...

Reply:
So for the key one, just needs to add a key to **both** places where you call the .map function. That should solve the warning. e.g. add a key on line 76.

Reply:
Yesterday, I forgot to add the key attribute to both div. hahaha. The warning is already solved.